File: frontend/libs/reporting-api/src/lib/model/create-run-payload.ts
import { ActorInfo } from './actor-info';
import { CommitInfo } from './commit-info';
import { CreateRunSuitePayload } from './create-run-suite-payload';

export interface CreateRunPayload {
  id: string;
  projectId: string;
  startDate: string;
  suites: CreateRunSuitePayload[];
  description?: string;
  pullRequestNumber?: number;
  commit?: CommitInfo;
  actor?: ActorInfo;
}

File: backend/OpenWright.Api/Reporting/Payloads/CreateRunPayload.cs
using OpenWright.Api.Runs.ValueObjects;

namespace OpenWright.Api.Reporting.Payloads;

public class CreateRunPayload
{
    public required Guid Id { get; init; }
    public required Guid ProjectId { get; init; }
    public required DateTimeOffset StartDate { get; init; }
    public required CreateRunSuitePayload[] Suites { get; init; }
    public string? Description { get; init; }
    public int? PullRequestNumber { get; init; }
    public CommitInfo? Commit { get; init; }
    public ActorInfo? Actor { get; init; }
}
